File: frontend/src/services/cartService.js
import api from "./api";

export const addToCart = async (
  userId,
  productId,
  quantity = 1
) => {
  return await api.post(`/cart/${userId}`, {
    productId,
    quantity,
  });
};

export const getCart = async (userId) => {
  return await api.get(`/cart/${userId}`);
};

export const updateCartQuantity = async (
  cartId,
  quantity
) => {
  return await api.put(`/cart/${cartId}`, null, {
    params: { quantity },
  });
};

export const removeFromCart = async (cartId) => {
  return await api.delete(`/cart/${cartId}`);
};

File: backend/src/main/java/com/ecommerce/backend/service/CartService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.request.AddToCartRequest;
import com.ecommerce.backend.dto.response.CartResponse;

import java.util.List;

public interface CartService {

    CartResponse addToCart(Long userId, AddToCartRequest request);

    List<CartResponse> getCart(Long userId);

    CartResponse updateQuantity(Long cartId, Integer quantity);

    void removeFromCart(Long cartId);
}